core/src/test/scala/unit/kafka/zk/ZooKeeperTestHarness.scala

```diff
@@ -52,6 +56,22 @@ abstract class ZooKeeperTestHarness extends JUnitSuite with Logging {
     if (zookeeper != null)
       CoreUtils.swallow(zookeeper.shutdown())
     Configuration.setConfiguration(null)
+    assertNoBrokerControllersRunning()
```

Comment:
Should we be checking these things in `@BeforeClass` and `@AfterClass`. That would make it cheaper. Also, can we measure the cost of `assertNoBrokerControllersRunning`?